src/TokenFuzzChecks.sol

```diff
@@ -0,0 +1,461 @@
+        vm.assume(to != address(0) && to != address(_token));
+        allowance = bound(allowance, 0, type(uint256).max - 1);
+        amount = bound(amount, allowance + 1, type(uint256).max);
+        address from = address(0xABCD);
+        deal(_token, from, TokenLike(_token).balanceOf(from) + amount, true);
+
+        vm.prank(from);
+        TokenLike(_token).approve(address(this), allowance);
+
+        vm.expectRevert(abi.encodePacked(_contractName, "/insufficient-allowance"));
+        TokenLike(_token).transferFrom(from, to, amount);
+    }
+
+    // ************************************************************************************************************
+    // PERMIT
+    // ************************************************************************************************************
+
+    function fuzzBulkCheckPermit(
+        address _token,
+        string memory _contractName,
+        uint128 privKey,
+        address to,
+        uint256 amount,
+        uint256 deadline,
+        uint256 nonce
+    ) internal {
+        fuzzCheckPermitEOA(_token, privKey, to, amount, deadline);
+        fuzzCheckPermitBadNonce(_token, _contractName, privKey, to, amount, deadline, nonce);
+        fuzzCheckPermitBadDeadline(_token, _contractName, privKey, to, amount, deadline);
+        fuzzCheckPermitPastDeadline(_token, _contractName, privKey, to, amount, deadline);
+        fuzzCheckPermitReplay(_token, _contractName, privKey, to, amount, deadline);
+    }
+
+    function fuzzCheckPermitEOA(
+        address _token,
+        uint248 privKey,
+        address to,
+        uint256 amount,
+        uint256 deadline
+    ) public {
+        uint256 privateKey = privKey;
+        if (deadline < block.timestamp) deadline = block.timestamp;
+        if (privateKey == 0) privateKey = 1;
+
+        address owner = vm.addr(privateKey);
+
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+            privateKey,
+            keccak256(
+                abi.encodePacked(
+                    "\x19\x01",
+                    TokenLike(_token).DOMAIN_SEPARATOR(),
+                    keccak256(abi.encode(PERMIT_TYPEHASH, owner, to, amount, 0, deadline))
+                )
+            )
+        );
+
+        vm.expectEmit(true, true, true, true);
+        emit Approval(owner, to, amount);
+        TokenLike(_token).permit(owner, to, amount, deadline, v, r, s);
+
+        assertEq(TokenLike(_token).allowance(owner, to), amount);
+        assertEq(TokenLike(_token).nonces(owner), 1);
+    }
+
+    function fuzzCheckPermitBadNonce(
+        address _token, 
+        string memory _contractName,
+        uint128 privateKey,
+        address to,
+        uint256 amount,
+        uint256 deadline,
+        uint256 nonce
+    ) public {
+        if (deadline < block.timestamp) deadline = block.timestamp;
+        if (privateKey == 0) privateKey = 1;
+        address owner = vm.addr(privateKey);
+        if (nonce == TokenLike(_token).nonces(owner)) nonce++;
```

Comment:
I guess this would look better as `vm.assume(nonce != TokenLike(_token).nonces(owner)))`. But again not sure if works like that.

Reply:
Changed